build/terraform/e2e/module.tf

```diff
@@ -14,6 +14,7 @@
 
 
 // Run:
+//  terraform init -backend-config="bucket=<YOUR_GCP_ProjectID>-e2e-infra-bucket-tfstate" -backend-config="prefix=terraform/state"
```

Comment:
out of curiousity, what happens if you don't provide these variables? does terraform error out in an obvious way, or does it default to something bad?

Reply:
If starting with a clean state, i.e. there is no tfstate file for this module in your dev environment, `terraform init` without those variables will succeed but the local root configuration directory for this module will be used as backend. But the following `terraform apply` will fail because the state doesn't match with the existing resources, unless the related resources (clusters, firewalls) are all deleted, which is basically starting from a brand new state.

Reply:
Good point. I think that's good enough, yeah.